src/library.js

```diff
@@ -3125,11 +3125,11 @@ addToLibrary({
 #endif
       onload(new Uint8Array(arrayBuffer));
       if (dep) removeRunDependency(dep);
-    }, (event) => {
+    }, (err) => {
       if (onerror) {
-        onerror();
+        onerror(err);
```

Comment:
I looked at the first few users of this and didn't see any that use the parameter - where did you see it used? Or was this for an out-of-tree usage?

Reply:
@Kojoley could you elaborate on what API you were using when you saw the difference above?    How where you using it exactly?

Reply:
```sh
mkdir link_test
echo ; > link_test.cpp
call emcc -fPIC -sSIDE_MODULE -Wall -g -c "link_test.cpp" -o "link_test.o"
call emcc -fPIC -sSIDE_MODULE "link_test.o" -o "link_test/foo.wasm"

echo int main() { return 0; } > main.cpp
call emcc -fPIC -sMAIN_MODULE -Wall -g -c "main.cpp" -o "main.o"
call emcc -fPIC -sMAIN_MODULE "main.o" "link_test/foo.wasm" -o "link_test.js"

node link_test.js
```

Reply:
Do you know where in that output it uses the parameter? Or is it hard to figure out? (I'd hope you can add `console.log(new Error().stack)` right where the parameter is received, to show the stack trace.)

Reply:
I'm not the person that can comfortably debug that for you, sorry. I don't write Javascript nor have a clue about emscripten internals. It took me a while already to get to this point.

Reply:
Thanks for the repro case.  I think this change is good but I will do a little more investigation here before landing so I (we) fully understand what is going on here.